inst/shiny/modules/tab_data/data_upload.R

```diff
@@ -52,32 +54,82 @@ data_upload_server <- function(id) {
 
     raw_data <- (
       reactive({
+        # Reset errors at the start of the reactive
+        file_loading_error(NULL)
+
         #' If no data is provided by the user, load dummy data
         if (is.null(input$data_upload$datapath) & is.null(datapath)) {
-          DUMMY_DATA
+          return(DUMMY_DATA)
+        }
+
+        if (is.null(input$data_upload$datapath)) {
+          log_info("Data upload module initialized with datapath: ", datapath)
+          paths <- datapath
+          filenames <- basename(datapath)
         } else {
-          if (is.null(input$data_upload$datapath)) {
-            log_info("Data upload module initialized with datapath: ", datapath)
-            final_datapath <- datapath
-          } else {
-            final_datapath <- input$data_upload$datapath
-          }
-
-          df <- tryCatch({
-            file_loading_error(NULL)
-            read_pk(final_datapath)
+          paths <- input$data_upload$datapath
+          filenames <- input$data_upload$name
+        }
+
+        # Iterate over files: Read and classify
+        read_results <- purrr::map2(paths, filenames, function(path, name) {
+          tryCatch({
+            # Attempt to read
+            data <- read_pk(path)
+            list(status = "success", data = data, name = name, type = "data")
           }, error = function(e) {
-            file_loading_error(e$message)
+            # If read_pk fails
+            # TODO: @Jana, check if settings file is loaded and then create settings override (719)
+            list(status = "error", message = e$message, name = name)
           })
+        })
 
-          if (is.null(file_loading_error())) {
-            log_success("User data loaded successfully.")
-            df
-          } else {
-            log_error("Error loading user data: ", file_loading_error())
+        # Process results
+        successful_loads <- purrr::keep(read_results, ~ .x$status == "success")
+        errors <- purrr::keep(read_results, ~ .x$status == "error")
+
+        # Handle Errors
+        if (length(successful_loads) == 0) {
+          error_msgs <- purrr::map_chr(errors, ~ paste0(.x$name, ": ", .x$message))
```

Comment:
style: Please use anonymous functions in place of formulas when possible.
```suggestion
          error_msgs <- purrr::map_chr(errors, \(x) paste0(x$name, ": ", x$message))
```